lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -205,44 +205,50 @@ struct ConvertLayoutOpConversion
     }
 #ifdef USE_ROCM
     if (auto mfmaLayout = layout.dyn_cast<MfmaEncodingAttr>()) {
-      SmallVector<Value> mfmaColIdx(4);
-      SmallVector<Value> mfmaRowIdx(16);
+      unsigned iNonKDim = mfmaLayout.getNonKDim();
+      assert(iNonKDim == 16 || iNonKDim == 32);
```

Comment:
Would it better to use if the assert statement is replaced by a `if` statement?

Reply:
Unfortunately, we do not have any fallback path for other values in nonKDim, so I used assert here.

Reply:
Note for reviewers:
@oplavsic pointed that he wants to refactor this part in his PR: https://github.com/ROCmSoftwarePlatform/triton/pull/247/files#diff-3eee00a71e5ff853d12cbbaaf7971b9d9372080b540b06386acab0e53530e9b9R214

I'll take his implementation